### PR TITLE
Adds a terraform example

### PR DESCRIPTION
**Pull request by jovial**
Possibly needs some work on the docs before merging. Currently just using what we have in skeleton.

### CODE REVIEW

**Comment by sjpb** [comment]
README for this should probably be updated by deleting a lot of it once #37 is merged.

File: environments/terraform-example/inventory/groups [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# FIXME: development cluster is currently not setup to use dedicated login nodes
```
Comment (sjpb) [inline environments/terraform-example/inventory/groups:1]: Logic assumes login != control so groups here and above comment need changing.

File: environments/terraform-example/inventory/groups [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+# This inventory should be changed to match production.
+[control:children]
+login
+
+[podman:children]
```
Comment (sjpb) [inline environments/terraform-example/inventory/groups:6]: Looks like this needs refreshing from the latest `layouts/everything`?

File: environments/terraform-example/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+inventory/hosts
```
Comment (sjpb) [inline environments/terraform-example/.gitignore:1]: What's the basis for gitignoring this? Just that it's really controlled by terraform?

**Review by sjpb — CHANGES_REQUESTED** [review]
README should really be updated/simplified after merging #37

**Comment by sjpb** [comment]
This needs a separate control node.

**Comment by sjpb** [comment]
Actually - I think it'll work just setting `openhpc_login_only_nodes: ''` in e.g `environments/terraform-example/inventory/group_vars/all/openhpc.yml`.

Not sure the groups are consistent with the `minimal` skelton though.

**Comment by brtknr** [comment]
I rebased this on master but didn't have any conflict. I have moved the terraform example to cookiecutter folder since the README in the cookiecutter references terraform.

**Review by jovial — COMMENTED** [review]
Looks good to me, but I can;t approve my own PR.

**Review by JohnGarbutt — APPROVED** [review]

**Comment by sjpb** [comment]
I haven't actually looked at this yet (sorry) but there's a fairly sophisticated terraform example [here](https://github.com/stackhpc/ansible-slurm-appliance/tree/nrel-workshop/environments/nrel-workshop/terraform) including cluster network definition, multiple compute node types, per-node overrides for images, and auto-generation of partition info from compute node types. All that came out of experience setting up NREL ...

**Comment by sjpb** [comment]
> ... since the README in the cookiecutter references terraform.

It won't after #85 but I'm not adverse to including TF as a default - otherwise getting started is quite hard!

**Pull request merged**